src/test/metrics_test.cpp

```diff
@@ -0,0 +1,780 @@
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <future>
+#include <memory>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../metric.hpp"
+#include "../metric_family.hpp"
+#include "../metric_registry.hpp"
+
+using namespace ovms;
+
+using testing::ContainsRegex;
+using testing::HasSubstr;
+using testing::Not;
+
+TEST(MetricsCommon, FamilyName) {
+    MetricRegistry registry;
+    auto family = registry.createFamily<MetricCounter>("name", "desc");
+    EXPECT_EQ(family->getName(), "name");
+}
+
+TEST(MetricsCommon, FamilyDesc) {
+    MetricRegistry registry;
+    auto family = registry.createFamily<MetricCounter>("name", "desc");
+    EXPECT_EQ(family->getDesc(), "desc");
+}
```

Comment:
Agree, I think that even the accessors are useless here (`getDesc` and `getName`) since we test those in `registry.collect()`. Will remove those, together with tests.